coriolis/tests/osmorphing/test_windows.py

```diff
@@ -0,0 +1,893 @@
+            '0',
+            None,
+            None,
+        ]
+
+        interfaces_reg_path = (windows.INTERFACES_PATH_FORMAT %
+                               mock.sentinel.key_name)
+        with self.assertLogs('coriolis.osmorphing.windows',
+                             level=logging.WARNING):
+            self.morphing_tools._compile_static_ip_conf_from_registry(
+                mock.sentinel.key_name)
+
+        self.conn.exec_ps_command.assert_has_calls([
+            mock.call(
+                "(((Get-ChildItem -Path '%s').Name | Select-String -Pattern "
+                "'[^\\\\]+$').Matches).Value" % interfaces_reg_path),
+            mock.call("(Get-ItemProperty -Path '%s\\interface1').EnableDHCP" %
+                      interfaces_reg_path),
+            mock.call("(Get-ItemProperty -Path '%s\\interface1').IPAddress" % (
+                interfaces_reg_path)),
+            mock.call("(Get-ItemProperty -Path '%s\\interface1').SubnetMask" %
+                      interfaces_reg_path),
+        ])
+
+    def test_compile_static_ip_conf_from_registry_no_static_ip(self):
+        self.conn.exec_ps_command.side_effect = [
+            'interface1',
+            '1',
+        ]
+        interfaces_reg_path = (windows.INTERFACES_PATH_FORMAT %
+                               mock.sentinel.key_name)
+
+        with self.assertLogs('coriolis.osmorphing.windows',
+                             level=logging.DEBUG):
+            self.morphing_tools._compile_static_ip_conf_from_registry(
+                mock.sentinel.key_name)
+        self.conn.exec_ps_command.assert_has_calls([
+            mock.call(
+                "(((Get-ChildItem -Path '%s').Name | Select-String -Pattern "
+                "'[^\\\\]+$').Matches).Value" % interfaces_reg_path),
+            mock.call("(Get-ItemProperty -Path '%s\\interface1').EnableDHCP" %
+                      interfaces_reg_path),
+        ])
+
+    def test_check_ips_info(self):
+        nics_info = [
+            {'ip_addresses': [mock.sentinel.ip_address]}
+        ]
+        ips_info = [
+            {'ip_address': mock.sentinel.ip_address}
+        ]
+        self.morphing_tools._check_ips_info(nics_info, ips_info)
+
+    def test_check_ips_info_different_ips(self):
+        nics_info = [
+            {'ip_addresses': [
+                mock.sentinel.ip_address, mock.sentinel.ip_address2]}
+        ]
+        ips_info = [
+            {'ip_address': mock.sentinel.ip_address}
+        ]
+
+        self.assertRaises(exception.OSMorphingException,
+                          self.morphing_tools._check_ips_info,
+                          nics_info, ips_info)
+
+    @mock.patch.object(
+        windows.BaseWindowsMorphingTools, '_get_cbslinit_scripts_dir'
+    )
+    @mock.patch.object(windows.utils, 'write_winrm_file')
+    @mock.patch.object(windows.json, 'dumps')
+    @mock.patch('base64.b64encode')
+    def test__write_static_ip_script(self, mock_b64encode, mock_json_dumps,
```

Comment:
This unit test can be written with only `write_winrm_file` being mocked like this:
```python
@mock.patch.object(windows.utils, 'write_winrm_file')
def test__write_static_ip_script(self, mock_write_winrm_file):
    nics_info = [
        {'ip_addresses': ["10.1.10.10"]}
    ]
    ips_info = [
        {'ip_address': "10.1.10.10"}
    ]
    self.morphing_tools._write_static_ip_script(
        'C:', nics_info, ips_info)

    contents = windows.STATIC_IP_SCRIPT_TEMPLATE % {
        'nics_info': base64.b64encode(
            json.dumps(nics_info).encode()).decode(),
        'ips_info': base64.b64encode(
            json.dumps(ips_info).encode()).decode(),
    }
    mock_write_winrm_file.assert_called_once_with(
        self.morphing_tools._conn,
        "C:\\LocalScripts\\01-static-ip-config.ps1",
        contents.encode())
```

Reply:
I suggest taking this as an example to adjust unnecessary mocking from the previous comments.